### PR TITLE
Remove unused functions reported by cppcheck

### PR DESCRIPTION
**Pull request by SmallJoker**
This change removes a few (but not all) unused functions.
Some unused helper functions were not removed due to their complexity and potential of future use.

A rough list of removed things:
 * unused debug streams -> use warningstream, verboststream etc
 * unused mapgen functions -> similar code already exists
 * content/packages.cpp -> was moved entirely to Lua

## To do

This PR is Ready for Review.

## How to test

1) Check whether it compiles.

### CODE REVIEW

File: src/chat.cpp [hunk 1]
```diff
@@ -220,11 +215,6 @@ void ChatBuffer::scrollBottom()
 	m_scroll = getBottomScrollPos();
 }
 
-void ChatBuffer::scrollTop()
-{
-	m_scroll = getTopScrollPos();
```
Comment (sfan5) [inline src/chat.cpp:225]: Although very short and trivial I'd prefer keeping this for consistency with `scrollBottom`

Comment (SmallJoker) [reply src/chat.cpp:225]: Change reverted.

**Review by sfan5 — APPROVED** [review]
Looks good.

**Comment by nerzhul** [comment]
nice !

**Review by nerzhul — APPROVED** [review]

**Pull request merged**